### DISCUSSION TITLE
Troubles getting Artisan to connect with Phidget HUB5000 / RTD Phidget 1200 / PT100

### DISCUSSION THREAD

**Discussion by MAKOMO** [artisan-roaster-scope/artisan#1753] [Q&A]
_Answering [a question raised over on home-barrista.com](https://www.home-barista.com/roasting/troubles-getting-artisan-to-connect-with-phidget-hub5000-rtd-phidget-1200-pt100-t96805.html#p1015893) (and by PM!)._

**What is the most simple setup connecting to Phidget modules on the wireless Phidget HUB5000?**

0) Install the [latest Artisan version](https://artisan-scope.org/download/)

1) Install the regular [Phidget driver](https://phidgets.com/) (not the standalone one)

2) on newer macOS: allow the Artisan.app the communication with devices on your local network (`System Settings >> Privacy & Security >> Local Network`). macOS should ask you for this on the first connect with a corresponding dialog, but you can set this manual as well.

<img width="717" alt="Screenshot 2024-11-26 at 08 10 23" src="https://github.com/user-attachments/assets/392ca017-6663-4ce6-8a9c-c36cf2d1d9fc">

3) don't set a Phidget server password on the HUB5000

_NOTE: there are two passwords, one to access the configurations on your device (set in the System tab on the device configuration page) and the other is the Phidgets server password (set in the Phidgets tab). Only the "Phidget server password" is relevant here._

4) configure Artisan to use the Phidget network discovery by keep the Host field and password fields empty in the Phidgets tab (menu `Config >> Device >> Phidgets`), just tick the Network flag. 

_NOTE: this auto discovery mode does not support passwords for obvious reasons._

5) instead of making use of the auto discovery as under 4) you can also set the Host to the hostname for example displayed in the Phidgets Control Panel under the Phdigets tab, "Remote Server". By default this is "hub5000.local". Using a hostname instead of an IP address has the advantage that it is stable over reboot of your router which might assign the HUB a different IP from time to time. The name is stable. In case you insist, you can also use the HUBs IP address there, something like 192.168.2.110. In any case you should check if that server is at all accessible from your mac by opening a terminal and test if ping works

```
# ping hub5000.local
PING hub5000.local (192.168.2.110): 56 data bytes
64 bytes from 192.168.2.110: icmp_seq=0 ttl=64 time=4.109 ms
64 bytes from 192.168.2.110: icmp_seq=1 ttl=64 time=3.517 ms
....
```

6) in case you want to have things super secure, and the test with an empty device password succeeded, set a device password on your Hub using its configuration page, Phidgets tab and use the enter the same in the corresponding field of the Phidgets tab (menu `Config >> Device`, Phidget tab)

**Configure for dual TMP1200**

Of course you need to properly configure Artisan for your connected Phidgets as well. The TMP1200 are not as trivial as other modules as they don't deliver valid data if the RTD type is not correctly set and they are one channel devices. Thus you need to configure the main Artisan device (menu `Config >> Devices`, ET/BT tab, Meter) as "Phidget TMP1200 1xRTD A" (don't confuse with the entry TMP1202 and the one with ending with a B) to read your BT. Add an extra device of the same type for your ET. You then need to "map" that ET signal to the ET channel of the main device by entering "Y3" in the field labeled "ET Y(x)" in the Smb ET/BT tab (menu `Config >> Device`, 3rd tab).

Finally, you need to tell Artisan which type of RTDs you have connected, a PT100 or a PT1000 and with which wiring, in the Phidget tab (menu `Config >> Device`, 4th tab). Note that there are two sets of configurations A and B, one for the device type ending in A and one for those ending in B. This allows you to use two different types of sensors at the same time, PT100 and PT1000.

**Comment by Teepeedeewhy** [comment]
Hi again,

I have another question if I could please ask for some help with?

Now that the 2x RTD's are connected and delivering ET/BT temperatures to Artisan, my next step is to control (ON/OFF) an electric heater. I have the physical devices correctly and with the Phidget control panel I can switch the heater ON/OFF using the duty cycle ON/OFF button. See image below:
<img width="1183" alt="Screenshot 2024-11-30 at 5 32 38 PM" src="https://github.com/user-attachments/assets/64bc4030-97c3-422b-83a3-903d09e9ce47">
I've been through many pages of documentation (I hope they were the right ones) but I'm unclear on which approach to take. I've tried using IO Command but the message is Failed to set(0,1). I've tried the VOUT command out(0,5) both with and without the serial number and port and that was unsuccessful.

I think I must be missing something - I was hoping someone could point me to some doco on how to convert the setting in the Phidget control panel port 0 channel 0 duty cycle ON of a Phidget HUB5000 into Artisan config commands?

Apologies if this has been asked previously - I searched many times and couldn't find it - but could just be me. If so please just send me a link.

Kind regards,

Tim.

**Reply by MAKOMO** [reply]
Phidget support is documented [here](https://artisan-scope.org/devices/phidgets/). Analog and Digital Output modules are discussed under [4. Analog/Digital Output](https://artisan-scope.org/devices/phidgets/#4-analogdigital-output). The section contains a link to the post [More Phidgets!](https://artisan-roasterscope.blogspot.com/2017/12/more-phidgets.html) which contains all the details.

There you find under "Output" the "PWM Command" which can be used to control the PWM output of a HUB. The `PWM Command`, `outhub(0,{})` will set the port 0 of your HUB to the PWM duty value [0-100%] as specified by the button/sliders event value (the `{}` place holder is replaced by that event value; you can also use a fixed 50 here). See screenshot below.

<img width="1115" alt="Screenshot 2024-11-30 at 11 12 55" src="https://github.com/user-attachments/assets/863aa870-8ab0-4e8d-8205-839f4872fe9c">

**Reply by Teepeedeewhy** [reply]
Many thanks. I will give it thorough read tomorrow and try those options. 

Thanks again

Tim

**Reply by Teepeedeewhy** [reply]
Thank you again MAKAMO!

At least this time I had already read the linked material you provided. I just didn’t realise I needed the PWM Command. Duty Cycle should have made that obvious I guess. 

Thanks for your help and all is working well now.
